### PR TITLE
Atomic-free JNI work

### PR DESCRIPTION
**Pull request by gacholio**
Implement FlushProcessWriteBuffers for AIX. Note this implementation
differs from Linux in that mlock is not called on the guard page (mlock
on AIX is only available to root-privilege users). 

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/vm/FlushProcessWriteBuffers.cpp [hunk 1]
```diff
@@ -37,50 +37,60 @@ extern "C" {
...
-	int mprotectrc = mprotect(addr, pageSize, PROT_READ | PROT_WRITE);
-	Assert_VM_true(0 == mprotectrc);
-	VM_AtomicSupport::add((UDATA*)addr, 1);
-	mprotectrc = mprotect(addr, pageSize, PROT_NONE);
-	Assert_VM_true(0 == mprotectrc);
-	omrthread_monitor_exit(vm->flushMutex);
-#else /* LINUX */
+	if (NULL != vm->flushFunction) {
+		((VOID (WINAPI*)(void))vm->flushFunction)();
+	}
```
Comment (charliegracie) [inline runtime/vm/FlushProcessWriteBuffers.cpp]: Should there be some runtime failure if the function does not exist? Similarly below for the mutex?

Comment (gacholio) [reply runtime/vm/FlushProcessWriteBuffers.cpp]: No, the point is to be able to acquire exclusive before it has been initialized, on the assumption that we're single-threaded.  See comment above.

Comment (gacholio) [reply runtime/vm/FlushProcessWriteBuffers.cpp]: If the initialize fails (can't allocate mutex/find function), the VM fails to start.

**Comment by DanHeidinga** [comment]
@charliegracie Are you good with this as well?

**Comment by DanHeidinga** [comment]
Depends on eclipse/omr#2729

File: runtime/vm/FlushProcessWriteBuffers.cpp (hunk 1 above)
Comment (charliegracie) [reply runtime/vm/FlushProcessWriteBuffers.cpp]: Sorry missed the comment.

**Comment by DanHeidinga** [comment]
Compile-only build was sufficient as this only affects AIX and is under #ifdefs that aren't yet enabled.

**Pull request merged**